website/static/js/share/results.js

```diff
@@ -1,126 +1,272 @@
-                                    return [
-                                        m('.badge.pointer', {onclick: function(){
-                                            utils.updateFilter(self.vm, 'tags:"' + tag + '"', true);
-                                        }}, tag.length < 50 ? tag : tag.substring(0, 47) + '...'),
-                                        ' '
-                                    ];
-                                };
-
-                                if (result.showAllTags || result.tags.length < 5) {
-                                    return $.map(result.tags, renderTag);
-                                }
-                                return m('span', [
-                                    $.map(result.tags.slice(0, 5), renderTag),
-                                    m('br'),
-                                    m('div', m('a', {onclick: function() {result.showAllTags = result.showAllTags ? false : true;}},'See All'))
-                                ]);
-                            }())))
+                        m('.pull-right',
+                          {style: {'text-align': 'right'}},
+                          m.component(Subjects, params)
+                        )
+                    )
                 ]),
                 m('br'),
-                m('br'),
-                m('div', [
-                    m('span', 'Released on ' + new $osf.FormattableDate(result.dateUpdated).local),
-                    m('span.pull-right', [
-                        m('img', {src: self.vm.ProviderMap[result.source].favicon, style: {width: '16px', height: '16px'}}),
-                        ' ',
-                        m('a', {onclick: function() {utils.updateFilter(self.vm, 'source:' + result.source);}}, self.vm.ProviderMap[result.source].long_name)
-                    ])
-                ])
+                m.component(Footer, params)
             ]),
             m('hr')
         ]);
-    };
-
-    // Uncomment for infinite scrolling!
-    // $(window).scroll(function() {
-    //     if  ($(window).scrollTop() === $(document).height() - $(window).height()){
-    //         utils.loadMore(self.vm);
-    //     }
-    // });
+
+    }
+};
+
+var TitleBar = {
+    view: function(ctrl, params) {
+        var result = params.result;
+        return m('span', {}, [
+            m('a[href=' + result.uris.canonicalUri + ']', ((result.title || 'No title provided'))),
+            m('br'),
+            m.component(Description, params)
+        ]);
+    }
+};
+
+/* Render the description of a single result. Will highlight the matched text */
+var Description = {
+    controller: function(vm) {
+        var self = this;
+        self.showAll = false;
+    },
+    view: function(ctrl, params) {
+        var result = params.result;
+        if ((result.description || '').length > 350) {
+            return m('p.readable.pointer', {
+                onclick: function() {
```

Comment:
... although this might be an exception because the text isn't really button-like. You might want to include a button that has the same expand/collapse functionality that is only visible on assistive technologies (Bootstrap provides a `.sr-only` helper for this).